AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSHTTPURLResponse.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// `AWSHTTPURLResponse` contains the response body and metadata associated with the response to an HTTP request.
+///
+/// When using `AWSAPIPlugin`, you can optionally type cast `HTTPURLResponse` to an instances of `AWSHTTPURLResponse`.
+/// The response body can be accessed from the `body: Data?` property. For example, when the `APIError` is an
+/// `.httpStatusError(StatusCode, HTTPURLResponse)`, then access the response body by type casting the response to
+/// an `AWSHTTPURLResponse` and retrieve the `body` field.
+/// ```swift
+/// if case let .httpStatusError(statusCode, response) = error, let awsResponse = response as? AWSHTTPURLResponse {
+///     if let responseBody = awsResponse.body {
+///         print("Response contains a response body")
```

Comment:
Nit to clarify that the response body is a Data object, and also to avoid compiler warnings in copy/pasted code :)

```suggestion
///         print("Response contains a \(responseBody.count) byte long response body")
```